data_schema/convert_value.py

```diff
@@ -104,7 +104,11 @@ def _preprocess_value(self, value, format_str, transform_case=None):
         """
         value = super(NumericConverter, self)._preprocess_value(value, format_str, transform_case=transform_case)
         if self.is_string(value):
-            value = self.NON_NUMERIC_REGEX.sub('', value) or None
+            if self.NON_NUMERIC_REGEX.search(value):
+                raise ValueError('non numeric characters found in numeric value')
+
+        if float(value) in [float('inf'), float('-inf')]:
+            raise ValueError('inf not a supported numeric value')
 
         return value
```

Comment:
it seems odd to me that the conversion method doesn't convert/cast the return value, but since that wasn't really in-scope for this problem/change, I left it as-is

Reply:
I see now where the conversion takes place